eventuate-adapter-vertx/src/main/scala/com/rbmhtechnology/eventuate/adapter/vertx/api/VertxAdapterConfig.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 - 2016 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.eventuate.adapter.vertx.api
+
+import akka.actor.ActorRef
+
+import scala.concurrent.duration.FiniteDuration
+
+sealed trait VertxAdapterConfig {
+  def id: String
+  def log: ActorRef
+}
+
+sealed trait VertxDispatcherConfig extends VertxAdapterConfig {
+  def endpointRouter: EndpointRouter
+}
+
+case class VertxPublisherConfig(id: String, log: ActorRef, endpointRouter: EndpointRouter) extends VertxDispatcherConfig
+case class VertxSenderConfig(id: String, log: ActorRef, endpointRouter: EndpointRouter, deliveryMode: DeliveryMode) extends VertxDispatcherConfig
```

Comment:
Hmmm, there is a `LogDispatcherConfig`. For symmetry reasons, why not renaming:

- `VertxDispatcherConfig` -> `BusDispatcherConfig`
- `VertxPublisherConfig` -> `BusPublisherConfig`
- `VertxSenderConfig` -> `BusSenderConfig`